### ISSUE TITLE
Perhaps be more specific about casing of `field-key`s

### ISSUE THREAD

**Issue by rylev** [WebAssembly/wasi-http#117]
[HTTP field keys are case *insensitive*](https://www.rfc-editor.org/rfc/rfc9110.html#section-5.1). However, `wasi-http` doesn't seem to specify that this should be the case when interacting with `fields`.

For example, `fields#get` likely should return values no matter the casing of the `field-key` passed in. This is how wasmtime's implementation currently works, but it doesn't seem specified anywhere that this *must* be the case. What's more, in wasmtime `fields#entries` returns tuples of the `field-key`/`field-value` pair where the `field-key` has the original casing. Is this the correct behavior or should hosts perhaps be expected to canonicalize these fields to lowercase instead?

**Comment by pchickey** [comment]
Good point. @guybedford what does node do for this behavior?

**Comment by guybedford** [comment]
Node.js automatically lowercases all header keys by the time they are provided to the user.

**Comment by lukewagner** [comment]
Yes, good point, we need to improve the specified behavior here.  I'd suggest the following:
* Specify that field-name equality (as used by `fields.get`, `fields.set`, etc) is case insensitive (as defined by RFC 9110)
* The `fields` resource remembers the original case of field names (exactly as passed to `set` or as came in over the wire), and this remembered-case is what goes out over the wire.  This is what the JS Fetch API specifies because it was found necessary in practice for various interop scenarios.
* Unlike with the JS Fetch API's `Headers.get()`, which returns always-lowered, and `Headers.entries()`, which returns always-sorted, specify that WASI HTTP's `fields.get` and `fields.entries` returns the original case in the over-the-wire order so that it's possible for content to know exactly what will go out over the wire.
